crates/nu-command/src/generators/generate.rs

```diff
@@ -170,6 +177,38 @@ used as the next argument to the closure, otherwise generation stops.
     }
 }
 
+fn get_initial_state(
+    initial: Option<Value>,
+    signature: &Signature,
+    span: Span,
+) -> Result<Value, ShellError> {
+    match initial {
+        Some(v) => Ok(v),
+        None => {
+            // the initial state should be referred from signature
+            if !signature.optional_positional.is_empty()
+                && signature.optional_positional[0].default_value.is_some()
+            {
+                Ok(signature.optional_positional[0]
+                    .default_value
+                    .clone()
+                    .expect("Already checked default value"))
+            } else {
+                Err(ShellError::GenericError {
+                    error: "The initial value is missing".to_string(),
+                    msg: "Missing initial value".to_string(),
+                    span: Some(span),
+                    help: Some(
+                        "Providing <initial> value to generate, or assigning default value to closure parameter"
```

Comment:
I think you can use imperative voice here.
```suggestion
                        "Provide an <initial> value as an argument to generate, or assign a default value to the closure parameter"
```